File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = list()
    result = 0
    for move in moves:
        for i in range(0, len(board)):
            doll = board[i][move - 1]
            if doll != 0:
                board[i][move - 1] = 0
                basket.append(doll)
                if len(basket) > 1:
                    if basket[-1] == basket[-2]:
                        basket.pop()
                        basket.pop()
                        result += 2
                break
    return result

File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> stack = new Stack();
        int answer = 0;
        
        for(int i = 0; i < moves.length; i++) {
            int move = moves[i] - 1;
            for(int j = 0; j < board.length; j++) {
                int doll = board[j][move];
                if(board[j][move] != 0){
                    board[j][move] = 0;
                    if (!stack.empty() && stack.peek() == doll) {
                        answer += 2;
                        stack.pop();
                    } else {
                        stack.push(doll);
                    }
                    break;
                }   
            }
        }
        
        return answer;
    }
}